src/main/kotlin/com/campspot/middleware/transactions/DAOManager.kt

```diff
@@ -0,0 +1,33 @@
+package com.campspot.middleware.transactions
+
+import com.campspot.dao.DAO
+import org.jdbi.v3.core.Handle
+import kotlin.reflect.KClass
+
+open class DAOManager(private vararg val daos: KClass<out DAO>) {
+  private val transaction = ThreadLocal<Handle>()
+  val daoInstances = ThreadLocal<HashMap<String, DAO>>()
+
+  fun setupWithTransaction(transaction: Handle): Handle {
+    this.transaction.set(transaction)
+    val daoMap = HashMap<String, DAO>()
+    daos.forEach { dao ->
```

Comment:
In each request thread we call this and make an instance of each DAO and attach it to the transaction